src/ucp/core/ucp_worker.c

```diff
@@ -1949,6 +1955,18 @@ ucs_thread_mode_t ucp_worker_get_thread_mode(uint64_t worker_flags)
     return UCS_THREAD_MODE_SINGLE;
 }
 
+static void ucp_warn_unused_uct_config(ucp_context_h context)
+{
+    ucs_config_cached_key_t *key_val;
+
+    ucs_list_for_each(key_val, &context->cached_key_list, list) {
+        if (key_val->used == 0) {
+            ucs_warn("unused uct configuration %s=%s",
+                     key_val->key, key_val->value);
```

Comment:
can you print all in same line, using a string_buffer, like it's done in https://github.com/openucx/ucx/blob/868fa808783bf1e0b8c770b54582f45d504c4bf0/src/ucs/config/parser.c#L1917